graphjoiner/declarative/__init__.py

```diff
@@ -23,6 +23,7 @@ def executor(root, mutation=None):
     return graphjoiner.executor(root_type, mutation=mutation_type)
 
 
+@attr.attrs
```

Comment:
`Type` isn't intended to be used as a value object, so I think we should get rid of the decorator.